src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -36,6 +36,47 @@ public function find($key, $default = null)
     }
 
     /**
+     * Determine if the given relation is loaded on collection models.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    public function relationLoaded($key)
+    {
+        // If collection is empty relations are still loaded, but nothing was found
+        if ($this->isEmpty()) {
+            return true;
+        }
```

Comment:
I guess I don't totally understand this method at all. It doesn't check if the relation is loaded on all models. It only checks the first model. I could easily create a Eloquent collection where only some of the models have the relationship loaded and this would still pass.